lib/src/git.rs

```diff
@@ -690,6 +692,22 @@ pub fn reset_head(
         git_repo.set_head_detached(new_git_commit_id)?;
         git_repo.reset(new_git_commit.as_object(), git2::ResetType::Mixed, None)?;
         mut_repo.set_git_head_target(RefTarget::normal(first_parent_id.clone()));
+    } else {
+        // Can't detach HEAD without a commit. Use random ref to nullify the HEAD.
+        // We can't set_head() an arbitrary unborn ref, so use reference_symbolic()
+        // instead. Git CLI appears to deal with that. It would be nice if Git CLI
+        // couldn't create a commit without setting a valid branch name.
+        if mut_repo.git_head().is_present() {
+            let random_bytes: [u8; 16] = rand::random();
+            let name = format!("{UNBORN_REF_NAMESPACE}{}", hex::encode(random_bytes));
+            git_repo.reference_symbolic("HEAD", &name, true, "unset HEAD by jj")?;
```

Comment:
Good point.

Since I originally used `.set_head("refs/heads/jj-unborn-...")`, I can't assume the ref is unborn. But "refs/jj" shouldn't be externally used, so we can freely delete it.

Reply:
Updated to use static `refs/jj/root` ref, and delete it as needed.